### PR TITLE
WIP: (v0.17.0) Enable JProfiling in Profiling Compilations

### PR DESCRIPTION
**Pull request by r30shah**
Use JProfiling infrastructure instead of JIT Profiling infrastructure for profiling compilations on all platform except x86-32 Bit platform.

Signed-off-by: Rahil Shah <rahil@ca.ibm.com>

### CODE REVIEW

**Comment by r30shah** [comment]
Opening up this PR to enable JProfiling from 0.17 release, please follow PR (https://github.com/eclipse/omr/pull/4370) for the discussion. 
FYI @andrewcraik @fjeremic

**Comment by pshipton** [comment]
@mstoodle fyi, a potential OMR change to the OpenJ9 0.17 release.

**Comment by pshipton** [comment]
Copying a summary provided by @r30shah of issues being investigated

Summarizing updates on the issues opened with JProfiling at the moment. 

1. https://github.com/eclipse/openj9/issues/7275 - Linux On Z On Jenkins Grinder fails 1/5
2. https://github.com/eclipse/openj9/issues/7299 - Linux On Z - I can locally reproduce this 1/100 failure rate
3. internal failure on x86, reproduction rate 1/100 (dup of 4).
4. https://github.com/eclipse/openj9/issues/7300 - On Windows - I am sure it is same issue as 3 
5. https://github.com/eclipse/openj9/issues/7276 - On Windows - On the Jenkins Grinder (https://github.com/eclipse/openj9/issues/7276#issuecomment-536364231) , does not fail in 35 runs.
I am actively looking into `1` and `2`, more specifically `1` which is marked blocker. As it fails 1/5. I first tried to reproduce this locally in 100 runs but I did not get any success. I also tried a 100x internal grinder. I am only able to reproduce this in the Jenkins grinder on `fyre` machine, and this morning, I got access to the internal `fyre` machine where it fails fairly and currently trying to reproduce issue there to investigate further. 
For `2`, I know the bad method which introduces the bug and I also have narrowed it down to the bad trees in the method. Basically we have introduced a pinning array store which stores internal pointer to that slot which is wrong. I need a log file with full tracing which can tell me which optimization causes the issue. But this one fails `1/100` and full tracing of the method suppresses the symptom and we do not reproduce this failure. 
I have a hunch that as all are failing in same DAA test, they are same issue so looking to fix the highly reproducible first and see if others are resolved with that. 
For the `5` I don't think so is the same issue but also it is not highly reproducible.
 
In summary, I see only one issue with JProfiling atm that needs urgent fixing and I am onto it now.

**Comment by pshipton** [comment]
As 0.17 was finalized, this can be closed.
